core/src/test/resources/sql/cast.iq

```diff
@@ -1707,4 +1707,16 @@ WHERE x BETWEEN 2 AND 4
 
 !ok
 
+# [CALCITE-6981] Runtime conversion of DECIMAL ARRAY to INT ARRAY fails with a ClassCastException
```

Comment:
will this work with nested arrays (INT ARRAY ARRAY)?
maybe you can add a test to check